contracts/YakRegistry.sol

```diff
@@ -1,40 +1,76 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
 
 import "./lib/Ownable.sol";
+import "./lib/EnumerableSet.sol";
+import "./YakStrategy.sol";
 
 /**
  * @notice YakRegistry is a list of officially supported strategies.
  * @dev DRAFT
  */
 contract YakRegistry is Ownable {
+    uint256 public strategiesLength;
+    mapping(address => uint256[]) public strategyIdsForDepositToken;
 
-    mapping(address => bool) public registeredStrategies;
+    mapping(address => bool) private registeredStrategies;
+    address[] private strategies;
+
+    struct StrategyInfo {
+        uint256 id;
+        address strategyAddress;
+        bool depositsEnabled;
```

Comment:
I think this is not the place for this, there's no guarantee that if we disable deposits on a strategy it'll be also mirrored here, perhaps the right path for this is to ask the strategy if they have depositsEnabled rather than storing here

Reply:
This is only used as return type for "strategyInfo". No state is stored in the registry and read directly from the strategy contract in "strategyInfo".